### PR TITLE
Adds chunk based serverside pagination

### PR DESCRIPTION
**Pull request by jschurse**
\+ Added PaginatedResponse Interface to DataService Responses
\+ Extended store to store all parameters needed for chunk loading
\+ Implemented server side pagination as dynamic chunk loading via 'page' parameter
\+ Added navigation to page of selected post (for when post is being opened via map)
\* renamed visiblePages => postsOnCurrentPage as a more concise name
\* removed unused imports

Please have a look if this solution is understandable.

### CODE REVIEW

**Comment by jschurse** [comment]
Fixes #88

**Comment by jschurse** [comment]
Fixes #89

File: app/src/utils/services/DataService.ts [hunk 1]
```diff
@@ -1,55 +1,40 @@
...
-                               radius
-                            }: {searchValues: string[],
-                                location: Location|undefined,
-                                radius: string
-                            }): Promise<any> {
         const query = new QueryBuilder();
         query.mustShouldMatch(searchValues.map((value) => [{key: 'title', value}, {key: 'categories', value}]).flat());
-
-        query.size(100);
+        query.from(from);
```
Comment (FiedlerJan) [inline app/src/utils/services/DataService.ts:36]: The compiler throws an error on this line.

Comment (jschurse) [reply app/src/utils/services/DataService.ts:36]: Please update your installed dependencies by running `npm i`

Comment (FiedlerJan) [reply app/src/utils/services/DataService.ts:36]: Bildschirmfoto_2020-06-25_20-22-30
I still get this error.

I am using an arch linux.

Comment (jschurse) [reply app/src/utils/services/DataService.ts:36]: Could you please delete your node_modules dir `rm -rf node_modules` and recreate it via `npm i` this seems to be a npm caching issue. We had similiar problems some days ago on the openArtBrowser.

Our CI/CD pipeline complete´s this build without problem: https://github.com/hochschule-darmstadt/einander-helfen/pull/91/checks?check_run_id=807562036

Comment (FiedlerJan) [reply app/src/utils/services/DataService.ts:36]: The clean install fixed the issue. Thanks.

File: app/src/store/store.ts [hunk 2]
```diff
@@ -43,19 +52,42 @@ const store: StoreOptions<RootState> = {
...
+        // Calculate the new from parameter to load the next resultSet chunk if necessary
+        const currentPageIndex = (page - 1) * state.hitsPerPage; // these hu-mons start counting their pages at 1...
+        const currentLoadedChunk = {
+          min: state.resultsFrom,
+          max: state.resultsFrom + state.resultSetSize - 1 // again with these hu-mons and their count beginning at 1...
+        };
+        if (! inChunk(currentPageIndex, currentLoadedChunk)) {
+          // Calculate the needed offset
+          // rounding off to the next multiple of our resultSetSize
+          const from = currentPageIndex - (currentPageIndex % state.resultSetSize);
```
Comment (FiedlerJan) [inline app/src/store/store.ts:87]: Modulo is used at this point.

Comment (jschurse) [reply app/src/store/store.ts:87]: Yes, to calculate the offset from where to load the posts from elastic (and this is working). 
We are looking for this one: https://github.com/hochschule-darmstadt/einander-helfen/blob/issue/88/app/src/views/ResultPage.vue#L278 where the same calculation should have happend.
This is a clear case of abstraction leaking and will be worked upon.

**Comment by jschurse** [comment]
> Something is not working correctly. If I go to page f.e. 11 and select the first post. I get redirected to page 1. When I go to page 11 the correct post is highlighted. It seems like a static integer of 10 is used as modulo.

I reafactored nearly all details of the chunk loading logic into to the store to encapsulate it and hide it from the resultpage.
The resultpage only watches to 'from' state propertry to decide if it has to dispatch the loading of the next posts.

Selecting a post in the list now doesn't result in a jump of page.

**Pull request merged**